lib/ansible/plugins/inventory/ovirt.py

```diff
@@ -0,0 +1,275 @@
+    short_description: oVirt inventory source
+    version_added: "2.10"
+    author: Bram Verschueren (@bverschueren)
+    requirements:
+      - ovirt-engine-sdk-python >= 4.2.4
+    extends_documentation_fragment:
+        - inventory_cache
+        - constructed
+    description:
+      - Get inventory hosts from the ovirt service.
+      - Requires a YAML file ending in 'ovirt.yml', 'ovirt4.yml', 'ovirt.yaml', 'ovirt4.yaml'.
+    options:
+      plugin:
+        description: the name of this plugin, it should always be set to 'ovirt' for this plugin to recognise it as it's own.
+        required: True
+        choices: ['ovirt']
+      ovirt_url:
+        description: URL to ovirt-engine API.
+        required: True
+        env:
+          - name: OVIRT_URL
+      ovirt_username:
+        description: ovirt authentication user.
+        required: True
+        env:
+          - name: OVIRT_USERNAME
+      ovirt_password:
+        description: ovirt authentication password.
+        required : True
+        env:
+          - name: OVIRT_PASSWORD
+      ovirt_cafile:
+        description: path to ovirt-engine CA file.
+        required: False
+      ovirt_query_filter:
+        required: false
+        description: dictionary of filter key-values to query VM's. See U(https://ovirt.github.io/ovirt-engine-sdk/master/services.m.html#ovirtsdk4\
+.services.VmsService.list) for filter parameters.
+      ovirt_hostname_preference:
+        required: false
+        description: list of options that describe the ordering for which hostnames should be assigned. See U(https://ovirt.github.io/ovirt-engin\
+e-api-model/master/#types/vm) for available attributes.
+        default: ['fqdn', 'name']
+        type: list
+'''
+
+EXAMPLES = '''
+# Sample content of ovirt.yml
+# plugin: ovirt
+# ovirt_url: http://localhost/ovirt-engine/api
+# ovirt_username: ansible-tester
+# ovirt_password: secure
+# ovirt_query_filter:
+#   search: 'name=myvm AND cluster=mycluster'
+#   case_sensitive: no
+#   max: 15
+# keyed_groups:
+#   - key: cluster
+#     prefix: 'cluster'
+# groups:
+#   dev: "'dev' in tags"
+# compose:
+#   ansible_host: devices["eth0"][0]
+'''
+
+import sys
+
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+from ansible.errors import AnsibleError, AnsibleParserError
+
+HAS_OVIRT_LIB = False
+
+try:
+    import ovirtsdk4 as sdk
+    HAS_OVIRT_LIB = True
+except ImportError:
+    HAS_OVIRT_LIB = False
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'ovirt'
+
+    def __init__(self):
+
+        super(InventoryModule, self).__init__()
+        self.connection = None
+
+    def _get_connection(self):
+        if self.connection is None:
+            self.connection = sdk.Connection(
+                url=self.ovirt_engine_url,
+                username=self.ovirt_engine_user,
+                password=self.ovirt_engine_password,
+                ca_file=self.ovirt_engine_cafile
```

Comment:
Add: `insecure=not bool(self.ovirt_engine_cafile),`

Reply:
Shall we simply assume (but document) that the absence of a CA file leads to an insecure connection ? Or shall I introduce an `insecure` option to control this ?

Reply:
you can add the option `insecure` but default would be as `not bool(self.ovirt_engine_cafile)`  as it is in https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/ovirt/ovirt_auth.py

Reply:
Right, I'll add an `insecure` option, but, imo it's rather dangerous that not specifying a CA automatically leads to an insecure connection (as is the case with ovirt_auth). 
I'd rather :
1) use provided CA (ignore `insecure`) or,
2) use system-wide if no CA provided (given `insecure` defaults to False)
3) a) fail (given `insecure` defaults to False)
    b) continue insecurely if user specifically sets `insecure=True`

whereas ovirt_auth:
1) use provided CA (ignore `insecure`) or,
2) a) continue insecurely if no CA is provided (`insecure` defaults to None, toggling to True if no CA is provided)
https://github.com/ansible/ansible/blob/0c597f4434d5dcde56d5b28ad62dc22bd0b4babf/lib/ansible/modules/cloud/ovirt/ovirt_auth.py#L261
b) fail if no CA is provided AND user specifically sets `insecure=False`

Reply:
okay good idea